tasks/components/layout/TwoColumn.tsx

```diff
@@ -21,49 +30,89 @@ type TwoColumnProps = {
 }: TwoColumnProps) => {
-  const [layoutState, setLayoutState] = useState<LayoutState>(initialLayoutState)
+  const ref = createRef<HTMLDivElement>()
+  const [fullWidth, setFullWidth] = useState(0)
+  const [leftWidth, setLeftWidth] = useState(0)
+  const [dragImage, setDragImage] = useState<HTMLElement>()
   const headerHeight = 64
+  const dividerHitBoxWidth = 24
+
+  useEffect(() => {
+    if (fullWidth === 0) {
+      setLeftWidth(baseLayoutPercentage * (ref.current?.clientWidth ?? 0))
+    }
+    setFullWidth(ref.current?.clientWidth ?? 0)
+  }, [ref.current?.clientWidth])
+
+  useEffect(() => setDragImage(document.createElement('div')), [])
+
+  // TODO Update this to be more clear and use all/better constraints
+  const calcPosition = (dragPosition: number) => {
+    let left = dragPosition
+    if (dragPosition < constraints.left.min) {
+      left = constraints.left.min
+    } else if (fullWidth - dragPosition - dividerHitBoxWidth < constraints.right.min) {
+      left = fullWidth - constraints.right.min
+    }
+    return left
+  }
+
+  const leftFocus = baseLayoutPercentage * fullWidth < leftWidth - dividerHitBoxWidth / 2
+
   return (
-    <div className={tw(`flex flex-row h-[calc(100vh_-_${headerHeight}px)]`)}>
-      <div className={tx('overflow-hidden', {
-        'w-7/12': layoutState === 'normal',
-        'w-[70%]': layoutState === 'smallSidebar',
-        'flex-1': layoutState === 'leftMaximized'
-      })}>
+    <div ref={ref} className={tw(`relative flex flex-row h-[calc(100vh_-_${headerHeight}px)]`)}>
+      <div
+        className={tw(`overflow-hidden`)}
+        style={{ width: leftWidth + 'px' }}
+      >
         <SimpleBarReact style={{ maxHeight: window.innerHeight - headerHeight }}>
-          {left(layoutState)}
+          {left(leftWidth)}
         </SimpleBarReact>
       </div>
       <div
-        className={tx(`relative flex my-4 rounded-lg bg-gray-300 w-0.5 justify-center`, { 'mr-4': layoutState === 'leftMaximized' })}>
+        draggable
+        onDragStart={event => {
+          event.dataTransfer.setDragImage(dragImage ?? document.createElement('div'), 0, 0)
+        }}
+        onDragEnd={event => {
+          if (disableResize) {
+            return
+          }
+          setLeftWidth(calcPosition(event.pageX))
+        }}
+        onDrag={event => {
+          if (disableResize) {
+            return
+          }
+          // Overscrolling to the right makes the value 0, so we block this
+          if (event.pageX !== 0) {
+            setLeftWidth(calcPosition(event.pageX))
+          }
+          event.preventDefault()
+        }}
+        className={tx(`relative h-full flex justify-center bg-white w-[${dividerHitBoxWidth}px]`, { 'cursor-col-resize': !disableResize })}
+      >
+          <div className={tw('bg-gray-300 my-4 rounded-lg w-0.5')} />
+          <div className={tw('absolute top-[50%] bg-gray-300 rounded-xl w-6 h-20 -translate-y-[50%]')} />
```

Comment:
A little bit smaller
```suggestion
          <div className={tw('absolute top-[50%] bg-gray-300 rounded-xl w-3 h-12 -translate-y-[50%]')} />
```

Reply:
Making the draggable smaller would increase the effort required by the user to drag

Reply:
Nevermind you were only concerned with the drag pill and not the entire draggable zone